ios/Exponent/Versioned/Core/Api/EXCategoryAction.m

```diff
@@ -0,0 +1,52 @@
+//
+//  EXCategoryAction.m
+//  Exponent
+//
+//  Created by smszymon on 16.10.2018.
+//  Copyright © 2018 650 Industries. All rights reserved.
+//
+
+#import "EXCategoryAction.h"
+
+@implementation EXCategoryAction
+
++ (instancetype)parseFromArray: (NSArray*) array {
```

Comment:
Do not parse `EXCategoryAction` from array. Pass a dictionary from JS and extract information from that dictionary, not using numbers as indexes, but string keys.

It would be much easier to read if such method would be written as follows:

```
- (UNNotificationAction *)notificationActionFromParams:(NSDictionary *)params
{
  if (params are of text input) {
    // configure and return UNTextInputNotificationAction
  }
  
  // configure and return UNNotificationAction
}
```

I think your approach is good and would suit well some long-lived objects (the subclassing hierarchy), but for such helper function I think this is an overkill